### ISSUE TITLE
Slough Council

### ISSUE THREAD

**Issue by balsingh84** [robbrad/UKBinCollectionData#1105]
### Name of Council

Slough Borough Council

### Example Address/Postcode

SL2 2EW

### Additional Information

https://www.slough.gov.uk/directory-record/32475/farnham-lane-homes-with-wheelie-bins

### Verification

- [X] I''ve checked the [wiki](https://github.com/robbrad/UKBinCollectionData/wiki/Councils#contents) and verified that my council has not been added
- [X] I''ve checked that a request for my council does not already exist in the [Issues tracker](https://github.com/robbrad/UKBinCollectionData/issues?q=is%3Aopen+is%3Aissue+label%3A"council+request")
- [X] I have provided a tested working address/postcode/UPRN with bin collections available, as well as a link to the council''s website
- [X] I understand that this project is run by volunteer contributors and completion depends on numerous factors - even with a request, we cannot guarantee if/when your council will get a script

**Comment by davida72** [comment]
Assign to me pls.

**Comment by davida72** [comment]
Non-standard. You don't enter an address but part of a street name.

_Directory guidance: enter the keyword for your road name for example: “Bath” for Bath Road. Do not include your house number or house name or a space after the road name._

**Comment by robbrad** [comment]
Suggested approach here (us selenium because URLs can change)

1. go to https://www.slough.gov.uk/bin-collections
2. Click search
3. Click View the A to Z of records
4. Use a user supplied value for their address
5. Match first letter and navigate to that letter of the alphabets, page
6. Match the address value supplied
7. Scrape and parse page contents

**Comment by davida72** [comment]
Nice find. That works. I'll do this.

**Comment by davida72** [comment]
This is still pretty messy. I'm navigating to the page that lists collection information and capturing it.

- The dates are in the format "[Every other Wednesday](https://www.slough.gov.uk/directory-record/33421/wavell-gardens-homes-with-wheelie-bins)"
- The correct Wednesday is established by finding out if you're on [route A or route B](https://www.slough.gov.uk/bin-collections/green-bin-wednesday-week-a-collection-dates)
- The A and B routes are a hand written list of dates
- There are revisions during bank holidays which can be established [here](https://www.slough.gov.uk/bins-waste-recycling/bank-holiday-collection-dates)

Here's [my branch](https://github.com/davida72/UKBinCollectionData/tree/slough). It gets bin types and "Every other Wednesday" but doesn't check for A/B route, parse dates and revise them as necessary.

Before progressing I'd like to sanity check the best approach.

**Comment by balajibalasa** [comment]
Please add Slough council to the data refresh.  Slough is  the largest council in Berkshire.

**Comment by davida72** [comment]
Agree that it's a good council to add, but it's a horrible one to process!

**Comment by robbrad** [comment]
Tested and working
